tests/kafka_check/test_min_isr.py

```diff
@@ -13,128 +13,148 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import mock
-import pytest
-
-from kafka_utils.kafka_check import status_code
-from kafka_utils.kafka_check.commands.min_isr import MinIsrCmd
-
-
-class TestMinIsrCommand(object):
-    REGULAR_TOPIC_NAME0 = 'topic_0'
-    REGULAR_TOPIC_NAME1 = 'topic_1'
-    TOPICS_STATE = {
-        REGULAR_TOPIC_NAME0: {
-            "version": 1,
-            "partitions": {
-                "0": {
-                    "controller_epoch": 64,
-                    "leader_epoch": 54,
-                    "replicas": [
-                        170396635,
-                        170398981
-                    ],
-                    "isr": [
-                        170398981
-                    ],
-                    "version": 1,
-                    "leader": 170396635
-                }
-            }
-        },
-        REGULAR_TOPIC_NAME1: {
-            "version": 1,
-            "partitions": {
-                "0": {
-                    "controller_epoch": 64,
-                    "leader_epoch": 189,
-                    "replicas": [
-                        170396635,
-                        170398981
-                    ],
-                    "isr": [
-                        170396635,
-                        170398981
-                    ],
-                    "version": 1,
-                    "leader": 170396635
-                }
-            }
-        },
+from kafka.common import PartitionMetadata
+
+from kafka_utils.kafka_check.commands.min_isr import get_min_isr
+from kafka_utils.kafka_check.commands.min_isr import process_metadata_response
+
+
+TOPICS_STATE = {
+    'topic_0': {
+        0: PartitionMetadata(
+            topic='topic_0',
+            partition=0,
+            leader=170396635,
+            replicas=(170396635, 170398981),
+            isr=(170398981,),
+            error=0,
+        ),
+    },
+    'topic_1': {
+        0: PartitionMetadata(
+            topic='topic_1',
+            partition=0,
+            leader=170396635,
+            replicas=(170396635, 170398981),
+            isr=(170396635, 170398981),
+            error=0,
+        ),
+    },
+}
+
+
+def test_get_min_isr_empty():
+    TOPIC_CONFIG_WITHOUT_MIN_ISR = {'version': 1, 'config': {'retention.ms': '86400000'}}
+    attrs = {'get_topic_config.return_value': TOPIC_CONFIG_WITHOUT_MIN_ISR}
+    zk_mock = mock.MagicMock(**attrs)
+
+    min_isr = get_min_isr(zk_mock, 'topic_0')
+
+    zk_mock.get_topic_config.assert_called_once_with('topic_0')
+    assert min_isr is None
+
+
+def test_get_min_isr():
+    TOPIC_CONFIG_WITH_MIN_ISR = {
+        'version': 1,
+        'config': {'retention.ms': '86400000'},
+        'min.insync.replicas': '3',
```

Comment:
The 'min.insync.replicas' should be in the 'config' dictionary. This means that our get_min_isr function is broken as well.